src/types/doughnutLabel.js

```diff
@@ -0,0 +1,190 @@
+import {Element, DoughnutController} from 'chart.js';
+import {drawLabel, measureLabelSize, translate, shouldFit, inLabelRange, getElementCenterPoint, measureLabelRectangle, setBorderStyle} from '../helpers';
+import {TAU, getAngleFromPoint} from 'chart.js/helpers';
+
+export default class DoughnutLabelAnnotation extends Element {
+
+  inRange(mouseX, mouseY, axis, useFinalPosition) {
+    return inLabelRange(
+      {x: mouseX, y: mouseY},
+      {rect: this.getProps(['x', 'y', 'x2', 'y2'], useFinalPosition), center: this.getCenterPoint(useFinalPosition)},
+      axis,
+      this.rotation
+    );
+  }
+
+  getCenterPoint(useFinalPosition) {
+    return getElementCenterPoint(this, useFinalPosition);
+  }
+
+  draw(ctx) {
+    const options = this.options;
+    if (!options.display || !options.content) {
+      return;
+    }
+    drawBackground(ctx, this);
+    ctx.save();
+    translate(ctx, this.getCenterPoint(), this.rotation);
+    drawLabel(ctx, this, options, this._fitRatio);
+    ctx.restore();
+  }
+
+  resolveElementProperties(chart, options) {
+    const meta = getDatasetMeta(chart, options);
+    if (!meta) {
+      return {};
+    }
+    const {controllerMeta, point, radius} = getControllerMeta(chart, options, meta);
+    let labelSize = measureLabelSize(chart.ctx, options);
+    const _fitRatio = getFitRatio(labelSize, radius);
+    if (shouldFit(options, _fitRatio)) {
+      labelSize = measureLabelSize(chart.ctx, options, _fitRatio);
```

Comment:
Does it need to be measured again? Maybe just a new function doing the fit? Simpler code and easier to test.

Reply:
Make sense... it's enough to multiply width and height by fitRatio....

Reply:
done